toy_vm.py

```diff
@@ -380,40 +464,123 @@ def load(
             )
 
         case _:
-            raise ValueError(f"Unknown {address_mode=}")
+            raise ValueError(f"Unknown {address_mode=} for load OP")
 
-    setattr(instance.REGISTERS, reg_name, val)
+    setattr(instance.REGISTERS, reg_name.value, val)
     if not ip_unmodified:
         instance.REGISTERS.IP += instance.word_in_bytes
 
 
+def store(
+    instance: CentralProcessingUnit,
+    source_reg_name: CPURegisterNames,
+    *,
+    dest_reg_name: Optional[CPURegisterNames] = None,
+    address_mode=AddressModes.IMMEDIATE,
+):
+    val = getattr(instance.REGISTERS, source_reg_name.value)
+
+    match address_mode:
+        case AddressModes.DIRECT_INDEXED:
+            val = val.to_bytes(length=instance.word_in_bytes)
+            address_bytes_val = instance.RAM[
+                instance.REGISTERS.IP : instance.REGISTERS.IP + instance.word_in_bytes
+            ]
+            address = int.from_bytes(address_bytes_val) + instance.REGISTERS.IX
+
+            instance.RAM[address : address + instance.word_in_bytes] = val
+
+            instance.REGISTERS.IP += instance.word_in_bytes
+        case AddressModes.REGISTER:
+            if not dest_reg_name:
+                raise ValueError("Missing destination register name")
+            setattr(instance.REGISTERS, dest_reg_name.value, val)
+
+        case _:
+            raise ValueError(f"Unknown {address_mode=} for store OP")
+
+
+def jump(instance: CentralProcessingUnit, flag_name: Optional[CPUFlagNames] = None):
+    if not flag_name or getattr(instance.FLAGS, flag_name.value):
+        load(instance, CPURegisterNames.IP, ip_unmodified=True)
+    else:
+        instance.REGISTERS.IP += instance.word_in_bytes
+
+
 @CentralProcessingUnit.register_instruction(InstructionCodes.NOP)
 def noop(instance: CentralProcessingUnit):
     pass
 
 
-@CentralProcessingUnit.bind_to_registers(
+@CentralProcessingUnit.register_instruction_variants(
```

Comment:
And here is what I was thinking with the instruction decorators... we can define multiple variations on instructions... add in new func to the specific VM get super fast that way.